vector-im/element-android
# #5417: Pinned location sharing

## Type of change


## Content


Implementation of the pinned location sharing. User can now select a location on the map different from its current location. User can also zoom to its current location using a dedicated button placed above the compass icon.

## Motivation and context


Closes #5417 

## Screenshots / GIFs


### User location sharing
<img src="https://user-images.githubusercontent.com/46314705/157465169-4e0cfb52-c7c4-43cd-8495-8efcaaeff8b5.gif" width=300 />

### Pinned location sharing
<img src="https://user-images.githubusercontent.com/46314705/157464541-2a6b42cb-90f2-4007-aea8-50cee8f16392.gif" width=300 />

## Tests



User location case
- Ensure location sharing option is enabled in "Settings" -> "Preferences"
- Go to a room or a direct message
- Click on the "add" icon
- Select the location icon
- Check you see the map with your current location
- Check you see the confirmation view with your avatar and color
- Click on the confirmation view
- Check the location is correctly shared in the timeline of the room/message

Pinned location case
- Ensure location sharing option is enabled in "Settings" -> "Preferences"
- Go to a room or a direct message
- Click on the "add" icon
- Select the location icon
- Check you see the map with your current location
- Move the map on another location
- Check the confirmation view has changed and corresponds to pin location option
- Check the pin on the map is now a generic pin instead of the avatar
- Click on the confirmation view
- Check the location is correctly shared in the timeline of the room/message

## Tested devices

- OS version(s): Android 11

## Checklist
